k8-cluster/on-prem/rke2/ansible/README.md

```diff
@@ -0,0 +1,142 @@
+# RKE2 Kubernetes Cluster Setup using Ansible
+
+This repository contains an Ansible playbook and associated configurations to set up an RKE2 Kubernetes cluster with a primary control plane, subsequent control plane nodes, agent nodes, and an etcd cluster. The playbooks are designed to install RKE2, configure nodes, and ensure the cluster is ready for use.
+
+## Table of Contents
+- [Prerequisites](#prerequisites)
+- [Inventory Setup](#inventory-setup)
+- [Playbooks](#playbooks)
+- [Templates](#templates)
+- [Kubernetes setup via Ansible playbook](#kubernetes-setup-via-Ansible-playbook)
+- [Configuration Details](#configuration-details)
+
+## Prerequisites
+
+Before running the playbooks, ensure the following:
+
+- Install Ansible on your local machine.
+- Ensure that nodes are accessible from your local machine for SSH authentication.
+- Verify that the SSH private key for the nodes is available on your local machine.
+- Supported operating systems include Ubuntu 24.04 and other Debian-based distributions.
+
+
+## Inventory Setup
+
+The `hosts.ini` file contains the configuration of your nodes. It is structured as follows:
+
+```
+# Primary control plane node. This is the main node where RKE2 will be installed first.
+[control_plane_primary]
+control-plane-1 ansible_host=<internal ip> ansible_user=root ansible_ssh_private_key_file=<pvt .pem file>
+
+# Subsequent control plane nodes. These are additional control plane nodes that will be added to the cluster after the primary.
+[control_plane_subsequent]
+control-subsequent_plane-1 ansible_host=<internal ip> ansible_user=root ansible_ssh_private_key_file=<pvt .pem file>
+
+# agent nodes. These nodes will run application workloads.
+[agents]
+agents-1 ansible_host=<internal ip> ansible_user=root ansible_ssh_private_key_file=<pvt .pem file>
+
+# etcd nodes. These nodes will run the etcd database for Kubernetes, which is responsible for storing the cluster's state.
+[etcd]
+etcd-1 ansible_host=<internal ip> ansible_user=root ansible_ssh_private_key_file=<pvt .pem file>
+
+# Global variables applied to all hosts in the inventory.
+[all:vars]
+# The version of RKE2 to install across the cluster.
+INSTALL_RKE2_VERSION=v1.28.9+rke2r1
+# The DNS domain name to use for the cluster.
+cluster_domain=<cluster name>
+# The path where RKE2 will be installed on each node.
+RKE2_PATH=/etc/rancher/rke2
+
+```
+
+- Update the `hosts.ini` file with the correct IP addresses, usernames, and paths to your private key files.
+
+### Playbooks
+
+#### `main.yaml`
```

Comment:
add playbook details in the README inside playbook directory which explains the usage deeply.

Here we should provide the detailed steps for                         Cluster creation , configuration etc.